VirtoCommerce.MarketingModule.Data/Services/MarketingSearchServiceImpl.cs

```diff
@@ -33,10 +33,17 @@ public GenericSearchResult<coreModel.Promotion> SearchPromotions(PromotionSearch
             using (var repository = _repositoryFactory())
             {
                 var query = repository.Promotions;
+
                 if (!string.IsNullOrEmpty(criteria.Store))
                 {
-                    query = query.Where(x => x.StoreId == criteria.Store);
+                    query = query.Where(x => x.Stores.Any(s => !x.Stores.Any() || s.StoreId == criteria.Store));
+                }
+
+                if (!criteria.StoreIds.IsNullOrEmpty())
+                {
+                    query = query.Where(x => !x.Stores.Any() || x.Stores.Any(s => criteria.StoreIds.Contains(s.StoreId)));
```

Comment:
Did you test this where clauses, I have some doubt about how the EF translated to SQL query? Maybe better will replace Any to Contains?

Reply:
I did, Any() does work in EF